packages/client/src/AdditionalApi.ts

```diff
@@ -123,104 +123,171 @@ export async function compressJSON(input: unknown): Promise<BodyInit> {
     throw new Error("Compression not support: this seems to be an old browser")
 }
 
-function offsetForMetaPointer(builder: FBBuilder, mp: LionWebJsonMetaPointer): number {
-    const languageOffset = builder.createString(mp.language)
-    const keyOffset = builder.createString(mp.key)
-    const versionOffset = builder.createString(mp.version)
-    return FBMetaPointer.createFBMetaPointer(builder, languageOffset, keyOffset, versionOffset)
-}
-
-export function encodeBulkImportToFlatBuffer(bulkImport: BulkImport): Uint8Array {
-    const builder = new FBBuilder(1024)
-    const containerByAttached: Record<string, string> = {}
+/**
+ * The `InterningContext` class is responsible for managing and interning strings, language versions,
+ * and meta-pointers, as needed when serializing to Protobuffer.
+ */
+class InterningContext {
```

Comment:
As this file is getting really long, I would prefer to put this class in a separate file.

Reply:
Done